core/src/rpc.rs

```diff
@@ -314,6 +316,21 @@ impl JsonRpcRequestProcessor {
         Ok(self.blocktree.get_confirmed_block(slot).ok())
     }
 
+    pub fn get_confirmed_blocks(
+        &self,
+        start_slot: Slot,
```

Comment:
hmm, I was just thinking here -- I don't know how to ensure `start_slot` is a root from the RPC API.   I essentially want to use first root in an epoch as`start_slot`.

Reply:
Maybe if `start_slot` is not a root we walk up the slots until a root is found or `end_slot` is reached?

Reply:
I think a solution would be to grab the RootedSlotIterator from 0 instead of from `start_slot`, and return as the first entry in the Vec the smallest rooted slot >= `start_slot`. (Which is basically the mirror of how I'm handling `end_slot`)

... assuming I've understood correctly, and the case you're trying to account for is when the slot that the epoch_schedule thinks is the first in the epoch didn't actually produce a block, and you're trying to find the first slot in the epoch that did.

Reply:
Whoops, I didn't see your 2nd comment. Yep, same idea.

Reply:
> ... epoch_schedule thinks is the first in the epoch didn't actually produce a block, and you're trying to find the first slot in the epoch that did.

Yep from the first to last slot in an epoch, gimme all them roots.  That's what I need

Reply:
Okay, updated